packages/app-project/src/components/AuthModal/components/RegisterForm/RegisterFormContainer.js

```diff
@@ -1,33 +1,153 @@
+import counterpart from 'counterpart'
 import { inject, observer } from 'mobx-react'
 import auth from 'panoptes-client/lib/auth'
+import { debounce } from 'lodash'
 import { func, shape } from 'prop-types'
 import React, { Component } from 'react'
+import en from './locales/en'
 
 import RegisterForm from './RegisterForm'
 
+const REMOTE_CHECK_DELAY = 1000
+counterpart.registerTranslations('en', en)
+
 class RegisterFormContainer extends Component {
   constructor() {
     super()
+    this.onChange = debounce(this.onChange.bind(this), REMOTE_CHECK_DELAY)
     this.onSubmit = this.onSubmit.bind(this)
+    this.validate = this.validate.bind(this)
+
+    this.state = {
+      error: ''
+    }
+  }
+
+  validate (values) {
+    const { email, emailConfirm, password, passwordConfirm, privacyAgreement } = values
+    let errors = {}
+    errors.emailConfirm = this.validateEmailConfirmation(email, emailConfirm)
+    errors.passwordConfirm = this.validatePassword(password, passwordConfirm)
+    errors.privacyAgreement = this.validatePrivacyAgreement(privacyAgreement)
+    return errors
```

Comment:
This is blocking `onSubmit` because Formik expects it to be an empty object for a valid form.

Reply:
There should also be a test that expects `validate` to return an empty object for valid forms.